doc/source/user/basics.subclassing.rst

```diff
@@ -31,6 +31,37 @@ things like array slicing.  The complications of subclassing ndarray are
 due to the mechanisms numpy has to support these latter two routes of
 instance creation.
 
+When to use subclassing
+=======================
+
+Besides the additional complexities of subclassing a NumPy array, subclasses
+can run into unexpected behaviour because some functions may convert the
+subclass to a baseclass and "forget" any additional information
+associated with the subclass.
```

Comment:
I would also note:

> Subclassing is also fragile, because it is easy to rely on inadvertent implementation details in NumPy that may change in the future.